R/main.R

```diff
@@ -1,118 +1,149 @@
-
-          scenObjects[[i]] <- currScenInfo
-          i <- i + 1
-        }
-      }
-    }
-
-  }
+  scenObjects <- Map(gen_ensemble_member,
+                     levels.AGROFOREST, levels.AGROFOREST_NONPASTURE, levels.CROPLAND,
+                     levels.LAGSHARE, levels.LINYEARS,
+                     seq_along(levels.AGROFOREST), aOutputDir) %>%
+    unlist(recursive=FALSE)
 
   # Loop over all scenario configurations and run the model
   foreach(obj = scenObjects) %dopar% {
     message("Starting simulation: ", obj$mFileName)
     run_model(obj)
   }
+  invisible(scenObjects)
+}
+
+#' Generate the ensemble members for a single set of parameters
+#'
+#' This generates one each of the Perfect, Lagged, and Linear scenario types
+#' using the input parameters.  The return value is a list of the three
+#' ScenarioInfo objects for the scenarios generated.
+#'
+#' @param agFor The logit exponent the ag/forest(?) nest
+#' @param agForNonPast The logit exponent for the non-pasture(?) nest
+#' @param crop The logit exponent for the crop nest
+#' @param share The share parameter for the lagged model
+#' @param linyears The number of years parameter for the linear model
+#' @param serialnum A serial number for generating unique file names
+#' @param outdir Name of the output directory
+#' @return List of three ScenarioInfo objects
+#' @keywords internal
+gen_ensemble_member <- function(agFor, agForNonPast, crop, share, linyears, serialnum, aOutputDir)
+{
+  ## Perfect expectations scenario
+  scenName <- getScenName(SCENARIO, "Perfect", NULL, agFor, agForNonPast, crop)
+
+  perfscen <- ScenarioInfo(aScenario = SCENARIO,
+                           aExpectationType = "Perfect",
+                           aLinearYears = NULL,
+                           aLaggedShareOld = NULL,
+                           aLogitUseDefault = FALSE,
+                           aLogitAgroForest = agFor,
+                           aLogitAgroForest_NonPasture = agForNonPast,
+                           aLogitCropland = crop,
+                           aScenarioName = scenName,
+                           aFileName = sprintf("Perf_%04d", serialnum),
+                           aOutputDir = aOutputDir)
+
+
+  ## Lagged scenario
+  scenName <- getScenName(SCENARIO, "Lagged", share, agFor, agForNonPast, crop)
+
+  lagscen <- ScenarioInfo(aScenario = SCENARIO,
+                          aExpectationType = "Lagged",
+                          aLinearYears = NULL,
+                          aLaggedShareOld = share,
+                          aLogitUseDefault = FALSE,
+                          aLogitAgroForest = agFor,
+                          aLogitAgroForest_NonPasture = agForNonPast,
+                          aLogitCropland = crop,
+                          aScenarioName = scenName,
+                          aFileName = sprintf("Lag_%04d", serialnum),
+                          aOutputDir = aOutputDir)
+
+
+  # Loop over all LINYEARS options and run Linear
```

Comment:
Is this the right comment for this one?

Reply:
I don't think so.  It looks like I forgot to update it when I reorganized this section.